controllers/devworkspace/solver/che_routing.go

```diff
@@ -188,15 +189,21 @@ func (c *CheRoutingSolver) provisionPodAdditions(objs *solvers.RoutingObjects, c
 		},
 		Resources: corev1.ResourceRequirements{
 			Limits: corev1.ResourceList{
-				corev1.ResourceMemory: resource.MustParse("256Mi"),
-				corev1.ResourceCPU:    resource.MustParse("0.5"),
+				corev1.ResourceMemory: resource.MustParse(constants.DefaultGatewayMemoryLimit),
+				corev1.ResourceCPU:    resource.MustParse(constants.DefaultGatewayCpuLimit),
 			},
 			Requests: corev1.ResourceList{
-				corev1.ResourceMemory: resource.MustParse("64Mi"),
-				corev1.ResourceCPU:    resource.MustParse("0.05"),
+				corev1.ResourceMemory: resource.MustParse(constants.DefaultGatewayMemoryRequest),
+				corev1.ResourceCPU:    resource.MustParse(constants.DefaultGatewayCpuRequest),
 			},
 		},
-	})
+	}
+
+	if cheCluster.Spec.DevEnvironments.GatewayContainer != nil {
+		_ = deploy.CustomizeContainer(gatewayContainer, cheCluster.Spec.DevEnvironments.GatewayContainer)
+	}
```

Comment:
Either `CustomizeContainer` should not return anything (currently it never returns anything but `nil`) or we shouldn't ignore the error here -- if a future update makes it sometimes return an error (e.g. by checking that request < limit) we probably won't notice a problem here until a bug is filed.


```suggestion
	if cheCluster.Spec.DevEnvironments.GatewayContainer != nil {
		if err := deploy.CustomizeContainer(gatewayContainer, cheCluster.Spec.DevEnvironments.GatewayContainer); err != nil {
			return err
		}
	}
```